core/src/main/java/com/scalar/db/storage/cosmos/ConcatenationVisitor.java

```diff
@@ -19,88 +19,89 @@
    */
   @Override
-  public void visit(BooleanValue value) {
-    values.add(String.valueOf(value.get()));
+  public void visit(BooleanColumn column) {
+    columns.add(String.valueOf(column.getBooleanValue()));
   }
 
   /**
-   * Sets the specified {@code IntValue} to the key string
+   * Sets the specified {@code IntColumn} to the key string
    *
-   * @param value a {@code IntValue} to be set
+   * @param column a {@code IntColumn} to be set
    */
   @Override
-  public void visit(IntValue value) {
-    values.add(String.valueOf(value.get()));
+  public void visit(IntColumn column) {
+    columns.add(String.valueOf(column.getIntValue()));
   }
 
   /**
-   * Sets the specified {@code BigIntValue} to the key string
+   * Sets the specified {@code BigIntColumn} to the key string
    *
-   * @param value a {@code BigIntValue} to be set
+   * @param column a {@code BigIntColumn} to be set
    */
   @Override
-  public void visit(BigIntValue value) {
-    values.add(String.valueOf(value.get()));
+  public void visit(BigIntColumn column) {
+    columns.add(String.valueOf(column.getBigIntValue()));
   }
 
   /**
-   * Sets the specified {@code FloatValue} to the key string
+   * Sets the specified {@code FloatColumn} to the key string
    *
-   * @param value a {@code FloatValue} to be set
+   * @param column a {@code FloatColumn} to be set
    */
   @Override
-  public void visit(FloatValue value) {
-    values.add(String.valueOf(value.get()));
+  public void visit(FloatColumn column) {
+    columns.add(String.valueOf(column.getFloatValue()));
   }
 
   /**
-   * Sets the specified {@code DoubleValue} to the key string
+   * Sets the specified {@code DoubleColumn} to the key string
    *
-   * @param value a {@code DoubleValue} to be set
+   * @param column a {@code DoubleColumn} to be set
    */
   @Override
-  public void visit(DoubleValue value) {
-    values.add(String.valueOf(value.get()));
+  public void visit(DoubleColumn column) {
+    columns.add(String.valueOf(column.getDoubleValue()));
   }
 
   /**
-   * Sets the specified {@code TextValue} to the key string
+   * Sets the specified {@code TextColumn} to the key string
    *
-   * @param value a {@code TextValue} to be set
+   * @param column a {@code TextColumn} to be set
    */
   @Override
-  public void visit(TextValue value) {
-    value.get().ifPresent(values::add);
+  public void visit(TextColumn column) {
+    column.getValue().ifPresent(columns::add);
   }
 
   /**
-   * Sets the specified {@code BlobValue} to the key string
+   * Sets the specified {@code BlobColumn} to the key string
    *
-   * @param value a {@code BlobValue} to be set
+   * @param column a {@code BlobColumn} to be set
    */
   @Override
-  public void visit(BlobValue value) {
-    // Use Base64 encoding
-    value
-        .get()
-        .ifPresent(b -> values.add(Base64.getUrlEncoder().withoutPadding().encodeToString(b)));
+  public void visit(BlobColumn column) {
+    if (!column.hasNullValue()) {
```

Comment:
Just out of curiosity: why not use `column.getValue().ifPresent()` as well as `visit(TextColumn column)`?

Reply:
Thanks for asking.
`column.getValue()` returns an `Optional<ByteBuffer>` but I need to obtain a `byte[]` which can only be obtained with `column.getBlobValueAsBytes()`

Reply:
I guess you can call `java.nio.ByteBuffer.array()` to get the byte array since the ByteBuffer instance seems byte-array backed one. But it instantiates a temporary ByteBuffer instance. So, either is fine with me 👍

Reply:
Yes, you are right, but I prioritize not instantiating the `ByteBuffer` over a syntax simplification.